Extension/src/LanguageServer/colorization.ts

```diff
@@ -0,0 +1,601 @@
+
+        // Next in priority would be a theme-specific user override of token color of the equivilent scope
+        if (colorTokenName) {
+            let settingValue: string = otherSettings.getCustomThemeSpecificColorToken(colorTokenName, themeName);
+            if (settingValue) {
+                if (baseCStyle) {
+                    baseCStyle.foreground = settingValue;
+                }
+                if (baseCppStyle) {
+                    baseCppStyle.foreground = settingValue;
+                }
+            }
+        }
+
+        // Next in priority would be a theme-specific user override of this scope in textMateRules
+        let textMateRules: TextMateRule[] = otherSettings.getCustomThemeSpecificTextMateRules(themeName);
+        this.findThemeStyleForScope(baseCStyle, baseCppStyle, scope, textMateRules);
+    }
+
+    private calculateThemeStyleForVsToken(tokenKind: TokenKind, scope: string, themeName: string, themeTextMateRules: TextMateRule[]): void {
+        let parts: string[] = scope.split(".");
+        let accumulatedScope: string = "";
+        for (let i: number = 0; i < parts.length; i++) {
+            accumulatedScope += parts[i];
+            this.calculateThemeStyleForScope(this.themeStyleCMap[tokenKind], this.themeStyleCppMap[tokenKind], accumulatedScope, themeName, themeTextMateRules);
+            this.calculateThemeStyleForScope(this.themeStyleCMap[tokenKind], null, accumulatedScope + ".c", themeName, themeTextMateRules);
+            this.calculateThemeStyleForScope(null, this.themeStyleCppMap[tokenKind], accumulatedScope + ".cpp", themeName, themeTextMateRules);
+            accumulatedScope += ".";
+        }
+    }
+
+    public reload(): void {
+        let otherSettings: OtherSettings = new OtherSettings(this.uri);
+        let themeName: string = otherSettings.colorTheme;
+        
+        // Enumerate through all extensions, looking for this theme.  (Themes are implemented as extensions - even the default ones)
+        // Crack open each package.json to check for a theme name patch
+        for (let i: number = 0; i < vscode.extensions.all.length; i++) {
+            let extensionPath: string = vscode.extensions.all[i].extensionPath;
+            let extensionPackageJsonPath: string = path.join(extensionPath, "package.json");
+            if (!fs.existsSync(extensionPackageJsonPath)) {
+                continue;
+            }
+            let packageJson: any = JSON.parse(fs.readFileSync(extensionPackageJsonPath).toString());
+            if (packageJson.contributes && packageJson.contributes.themes) {
+                for (let j: number = 0; j < packageJson.contributes.themes.length; j++) {
+                    let themeId: string = packageJson.contributes.themes[j].id;
+                    let themeLabel: string = packageJson.contributes.themes[j].label;
+                    if (themeId === themeName || themeLabel === themeName) {
```

Comment:
Is this for loop only looking for selected theme? Can you do a "find" or a "filter" on the array? https://stackoverflow.com/questions/42580100/typescript-take-object-out-of-array-based-on-attribute-value